db_sql.go

```diff
@@ -542,6 +542,64 @@ func (x *sqlUserStoreDB) UnarchiveIdentity(userId UserId) error {
 	return nil
 }
 
+func (x *sqlUserStoreDB) SetUserStats(userId UserId, action string) error {
+	var tx *sql.Tx
+	var err error
+	var update sql.Result
+
+	if tx, err = x.db.Begin(); err != nil {
+		return fmt.Errorf("could not begin transaction: %v", err)
+	}
+	defer tx.Rollback()
+
+	if action == UserStatActionLogin {
+		update, err = tx.Exec(`INSERT INTO authuserstats (user_id, last_login_date)
+							   VALUES ($1, NOW())
+							   ON CONFLICT (user_id) 
+							   DO UPDATE SET last_login_date = NOW()`, userId)
+	}
+	if action == UserStatActionEnable {
+		update, err = tx.Exec(`INSERT INTO authuserstats (user_id, enabled_date, disabled_date)
+							   VALUES ($1, NOW(), NULL)
+							   ON CONFLICT (user_id) 
+							   DO UPDATE SET enabled_date = NOW(), disabled_date = NULL`, userId)
+	}
+	if action == UserStatActionDisable {
+		update, err = tx.Exec(`INSERT INTO authuserstats (user_id, enabled_date, disabled_date)
+							   VALUES ($1, NULL, NOW())
+							   ON CONFLICT (user_id)
+							   DO UPDATE SET enabled_date = NULL, disabled_date = NOW()`, userId)
+	}
+	if err != nil {
```

Comment:
I would add an else with an error in here, so that if we accidentally miss handling a new action type, we'll see it.